### DISCUSSION TITLE
Offline/Online Support - Duplicate Id's

### DISCUSSION THREAD

**Discussion by hachadwick** [Mimetis/Dotmim.Sync#628] [Q&A] [answered]
Before I begin, let me say, this is an amazing library and the documentation is some of the best I've seen. 
I have a .NET project that has a web app (Blazor) and a desktop client (WPF). The web app is hosted in Azure along with the Azure SQL database that stores our data. Our desktop app, which is a WPF app, will have a local DB (most likely SQLite). Bost of these apps, web and desktop will perform generally the same functions. I am using a local DB on the desktop client to have much better response time and to have a offline mode for the user. I want the desktops database to sync with the Azure database. Your library does this perfectly but I can't figure out one aspect. 
To give some clarity, this app is not a public app, but is a internal application that our company uses to do our work. Here is the use case I am trying to resolve:

1. At the beginning, both Azure SQL DB and the SQLite DB are synced.
2. Then a web user creates a entry in the DB through the Blazor app.
3. Then a desktop user creates a entry with the desktop app and syncs with the Azure SQL DB.

In this scenario, the content on Client is overridden with the server because I have it set to Server wins in a conflict. This same issue occurs when it is set to client wins, just in the other direction. The root of the issue is, I believe, that each user, in their separate environments, created a entry in the DB's with the same ID. Thus, when they synced from the client, the entry on the losing side is removed. 

How should I handle this case? I notice there is a conflict resolution in your library, which is amazing but is there a way to accept BOTH entries? I imagine this could work like visual studio when there is a merge conflict: both entries are valid thus both are accepted. I sure appreciate any advice or guidance on this matter.

**Comment by Mimetis** [comment, marked as answer]
Same iDS means an Update conflict to resolve.
If you want to avoid this kind of situation, you should use GUID as primary keys.
That way, you are sure that you will not have sync conflict when inserting rows on both side

More on the subject here : https://github.com/Mimetis/Dotmim.Sync/issues/590

**Comment by hachadwick** [comment]
Awesome. Man, you are amazing. That works great. I appreciate it.
